src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -186,203 +183,134 @@ public bool TryGetData(string column, out object value)
+            foreach (ISqlMigrationAssessmentResult r in assessmentResults)
             {
-                var migrationResult = r as ISqlMigrationAssessmentResult;
-                if (migrationResult == null)
-                {
-                    continue;
-                }
-
-                var targetName = !string.IsNullOrWhiteSpace(migrationResult.DatabaseName)
-                                     ? $"{target.ServerName}:{migrationResult.DatabaseName}"
-                                     : target.Name;
-                var ruleId = migrationResult.FeatureId.ToString();
-
-                var item = new MigrationAssessmentInfo()
-                {
-                    CheckId = r.Check.Id,
-                    Description = r.Check.Description,
-                    DisplayName = r.Check.DisplayName,
-                    HelpLink = r.Check.HelpLink,
-                    Level = r.Check.Level.ToString(),
-                    TargetName = targetName,
-                    DatabaseName = migrationResult.DatabaseName,
-                    ServerName = migrationResult.ServerName,
-                    Tags = r.Check.Tags.ToArray(),
-                    TargetType = target.Type,
-                    RulesetName = Engine.Configuration.DefaultRuleset.Name,
-                    RulesetVersion = Engine.Configuration.DefaultRuleset.Version.ToString(),
-                    RuleId = ruleId,
-                    Message = r.Message,
-                    AppliesToMigrationTargetPlatform = migrationResult.AppliesToMigrationTargetPlatform.ToString(),
-                    IssueCategory = "Category_Unknown"
-                };
-
-                if (migrationResult.ImpactedObjects != null)
-                {
-                    ImpactedObjectInfo[] impactedObjects = new ImpactedObjectInfo[migrationResult.ImpactedObjects.Count];
-                    for (int i = 0; i < migrationResult.ImpactedObjects.Count; ++i)
-                    {
-                        var impactedObject = new ImpactedObjectInfo()
-                        {
-                            Name = migrationResult.ImpactedObjects[i].Name,
-                            ImpactDetail = migrationResult.ImpactedObjects[i].ImpactDetail,
-                            ObjectType = migrationResult.ImpactedObjects[i].ObjectType
-                        };
-                        impactedObjects[i] = impactedObject;
-                    }
-                    item.ImpactedObjects = impactedObjects;
-                }
-
-                result.Add(item);
+                assessmentResultLookup.Add(CreateAssessmentResultKey(r as ISqlMigrationAssessmentResult), r as ISqlMigrationAssessmentResult);
             }
-            return result;
+            ISqlMigrationAssessmentModel assessmentResultBig = await engine.GetTargetAssessmentResultsList(System.Threading.CancellationToken.None);
```

Comment:
I not sure what "Big" means in the name.  Perhaps a comment would help, or simply name assessmentResult?